src/unity/lib/visualization/summary_view.cpp

```diff
@@ -0,0 +1,100 @@
+#include "summary_view.hpp"
+
+#include <unity/lib/visualization/vega_data.hpp>
+#include <unity/lib/visualization/vega_spec.hpp>
+#include <math.h>
+
+using namespace turi;
+using namespace turi::visualization;
+
+summary_view_transformation_output::summary_view_transformation_output(const std::vector<std::shared_ptr<transformation_output>> outputs)
+  : m_outputs(outputs) {
+}
+
+std::string summary_view_transformation_output::vega_column_data(bool sframe) const {
+  std::stringstream ss;
+  ss << "{\"a\": " << std::to_string(index) << ",";
+  std::string title = extra_label_escape(column_names[index]);
+  ss << "\"title\": " << title << ",";
+  ss << "\"num_row\": " << size << ",";
+
+  switch (column_types[index]) {
+    case flex_type_enum::INTEGER:
+    case flex_type_enum::FLOAT:
+    case flex_type_enum::STRING:
+    {
+      auto h_result = std::dynamic_pointer_cast<sframe_transformation_output>(m_outputs[0]);
+      ss << h_result->vega_summary_data();
+      ss << "}";
+      break;
+    }
+    default:
+      throw std::runtime_error("Unexpected dtype. SFrame plot expects int, float or str dtypes.");
+  }
+  return ss.str();
+}
+
+summary_view_transformation::summary_view_transformation(const std::vector<std::shared_ptr<transformation_base>> transformers)
+  : m_transformers(transformers) {
+    // check some assumptions that make fuse work
```

Comment:
Let's make the terminology consistent here - it doesn't make sense that to get a summary view transformation, you "fuse" a "collection" of transformers. (Though that is what this code evolved from.)

Maybe we should have a summary_view_transformation take a collection of transformers in its constructor, instead of putting a method `fuse` on the collection itself?

Reply:
Potentially, I'll change that in another PR if that's alright, this one is getting big